### PR TITLE
Do not emit Array.Empty for an empty params call if the params is an array of pointers.

### PR DESCRIPTION
**Pull request by YairHalberstadt**

### CODE REVIEW

File: src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs [hunk 1]
```diff
@@ -907,31 +907,31 @@ bool isLambdaConversion(BoundExpression expr)
             // if it's available.  However, we also disable the optimization if we're in an expression lambda, the 
             // point of which is just to represent the semantics of an operation, and we don't know that all consumers
             // of expression lambdas will appropriately understand Array.Empty<T>().
-            if (arrayArgs.Length == 0 && !_inExpressionLambda)
+            // We disable it for pointer types as well, since they cannot be used as Type Arguments.
+            if (arrayArgs.Length == 0
+                && !_inExpressionLambda
+                && paramArrayType is ArrayTypeSymbol ats // could be false if there's a semantic error, e.g. the params parameter type isn't an array
+                && !ats.ElementType.IsPointerType())
```
Comment (AlekseyTs) [inline src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs]: >!ats.ElementType.IsPointerType() [](start = 19, length = 32)

What about the case of a nested array of pointers? Is that allowed? Please add a test.

**Comment by AlekseyTs** [comment]
Done with review pass (iteration 1)

File: src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs (hunk 1 above)
Comment (YairHalberstadt) [reply src/Compilers/CSharp/Portable/Lowering/LocalRewriter/LocalRewriter_Call.cs]: Done

**Comment by AlekseyTs** [comment]
@dotnet/roslyn-compiler For the second review.

**Comment by AlekseyTs** [comment]
@YairHalberstadt Thanks for the contribution. Would it be Ok if I squash commits in the process of merging this PR?

**Comment by YairHalberstadt** [comment]
Sure

**Pull request merged**